gallery_dl/extractor/facebook.py

```diff
@@ -0,0 +1,454 @@
+# -*- coding: utf-8 -*-
+
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 2 as
+# published by the Free Software Foundation.
+
+"""Extractors for https://www.facebook.com/"""
+
+from .common import Extractor, Message
+from .. import text, exception
+
+BASE_PATTERN = r"(?:https?://)?.*?facebook\.com"
+
+
+class FacebookExtractor(Extractor):
+    """Base class for Facebook extractors"""
+    category = "facebook"
+    root = "https://www.facebook.com"
+    directory_fmt = ("{category}", "{username}", "{title} ({set_id})")
+    filename_fmt = "{id}.{extension}"
+    archive_fmt = "{id}.{extension}"
+
+    set_url_fmt = root + "/media/set/?set={set_id}"
+    photo_url_fmt = root + "/photo/?fbid={photo_id}&set={set_id}"
+
+    def __init__(self, match):
+        Extractor.__init__(self, match)
+        self.match = match
+
+    def _init(self):
+        self.session.headers["Accept"] = "text/html"
+        self.session.headers["Sec-Fetch-Mode"] = "navigate"
+
+        self.fallback_retries = self.config("fallback-retries", 2)
+        self.sleep_429 = self.config("sleep-429", 5)
```

Comment:
A `sleep_429` equivalent is already defined in the base Extractor class
https://github.com/mikf/gallery-dl/blob/d1ad97ae0c1417bbcabb84fba13ecd514ba11223/gallery_dl/extractor/common.py#L47
https://github.com/mikf/gallery-dl/blob/d1ad97ae0c1417bbcabb84fba13ecd514ba11223/gallery_dl/extractor/common.py#L333-L335

You can access it like [here](https://github.com/mikf/gallery-dl/blob/d1ad97ae0c1417bbcabb84fba13ecd514ba11223/gallery_dl/extractor/common.py#L233-L237).